lib/plugins/aws/package/compile/events/cognitoUserPool/index.js

```diff
@@ -103,12 +103,16 @@ class AwsCompileCognitoUserPoolEvents {
 
       const userPoolLogicalId = this.provider.naming.getCognitoUserPoolLogicalId(poolName);
 
+      const DependsOn = _.map(currentPoolTriggerFunctions, (value) => this
+        .provider.naming.getLambdaLogicalId(value.functionName));
+
       const userPoolTemplate = {
         Type: 'AWS::Cognito::UserPool',
```

Comment:
I'm not sure if it is wise to create the user pool for the user. There are lots of other User Pool settings that have to be made available in that case (e.g. password complexity, MFA, any many more). The standard settings are not usable in a production environment.